include/boost/geometry/formulas/sjoberg_intersection.hpp

```diff
@@ -0,0 +1,372 @@
+                             CT & lon, CT & lat,
+                             Spheroid const& spheroid)
+    {
+        // coordinates in radians
+
+        // TODO - handle special cases like degenerated segments, equator, poles, etc.
+
+        CT const c0 = 0;
+        CT const c1 = 1;
+        CT const c2 = 2;
+
+        CT const pi = math::pi<CT>();
+        CT const pi_half = pi / c2;
+        CT const f = detail::flattening<CT>(spheroid);
+        CT const one_minus_f = c1 - f;
+        CT const e_sqr = f * (c2 - f);
+        
+        CT const sin_alpha1 = sin(alpha1);
+        CT const sin_alpha2 = sin(alpha2);
+
+        CT const tan_beta1 = one_minus_f * tan(lat1);
+        CT const tan_beta2 = one_minus_f * tan(lat2);
+        CT const beta1 = atan(tan_beta1);
+        CT const beta2 = atan(tan_beta2);
+        CT const cos_beta1 = cos(beta1);
+        CT const cos_beta2 = cos(beta2);
+        CT const sin_beta1 = sin(beta1);
+        CT const sin_beta2 = sin(beta2);
+
+        // Clairaut constants (lower-case in the paper)
+        int const sign_C1 = math::abs(alpha1) <= pi_half ? 1 : -1;
+        int const sign_C2 = math::abs(alpha2) <= pi_half ? 1 : -1;
+        // Cj = 1 if on equator
+        CT const C1 = sign_C1 * cos_beta1 * sin_alpha1;
+        CT const C2 = sign_C2 * cos_beta2 * sin_alpha2;
+
+        CT const sqrt_1_C1_sqr = math::sqrt(c1 - math::sqr(C1));
+        CT const sqrt_1_C2_sqr = math::sqrt(c1 - math::sqr(C2));
+
+        // handle special case: segments on the equator
+        bool const on_equator1 = math::equals(sqrt_1_C1_sqr, c0);
+        bool const on_equator2 = math::equals(sqrt_1_C2_sqr, c0);
+        if (on_equator1 && on_equator2)
+        {
+            return false;
+        }
+        else if (on_equator1)
+        {
+            CT const dL2 = d_lambda_e_sqr(sin_beta2, c0, C2, sqrt_1_C2_sqr, e_sqr);
+            CT const asin_t2_t02 = asin(C2 * tan_beta2 / sqrt_1_C2_sqr);
+            lat = c0;
+            lon = lon2 - asin_t2_t02 + dL2;
+            return true;
+        }
+        else if (on_equator2)
+        {
+            CT const dL1 = d_lambda_e_sqr(sin_beta1, c0, C1, sqrt_1_C1_sqr, e_sqr);
+            CT const asin_t1_t01 = asin(C1 * tan_beta1 / sqrt_1_C1_sqr);
+            lat = c0;
+            lon = lon1 - asin_t1_t01 + dL1;
+            return true;
+        }
+
+        CT const t01 = sqrt_1_C1_sqr / C1;
+        CT const t02 = sqrt_1_C2_sqr / C2;
+
+        CT const asin_t1_t01 = asin(tan_beta1 / t01);
+        CT const asin_t2_t02 = asin(tan_beta2 / t02);
+        CT const t01_t02 = t01 * t02;
+        CT const t01_t02_2 = c2 * t01_t02;
+        CT const sqr_t01_sqr_t02 = math::sqr(t01) + math::sqr(t02);
+
+        CT t = tan_beta1;
+        int t_id = 0;
+
+        // find the initial t using simplified spherical solution
+        // though not entirely since the reduced latitudes and azimuths are spheroidal
+        // paper 2007
```

Comment:
Instead of paper 2007, paper, etc it is more standard to use something like [Sjoberg02].